### PR TITLE
Chunk features CI

### PR DESCRIPTION
**Pull request by Manishearth**
Based off of https://github.com/unicode-org/icu4x/pull/3741/

Fixes https://github.com/unicode-org/icu4x/issues/3675

https://github.com/unicode-org/icu4x/pull/3745 fixed CI but I'd like to improve features CI time further.

<s>Needs https://github.com/frewsxcv/cargo-all-features/pull/47</s>

### CODE REVIEW

**Comment by Manishearth** [comment]
(note that for the first CI run, subtract a minute from all features CI since it also needed to build cargo-all-features. I've added caching)

File: tools/make/tests.toml [hunk 1]
```diff
@@ -4,19 +4,24 @@
...
+category = "ICU4X Development"
+install_crate = { crate_name = "cargo-all-features", binary = "cargo-check-all-features", test_arg = ["--help"] }
+command = "cargo"
+args = ["check-all-features", "--n-chunks", "3", "--chunk", "${@}"]
+
 [tasks.check-all-features]
 description = "Build all permutations of all features"
 category = "ICU4X Development"
 install_crate = { crate_name = "cargo-all-features", binary = "cargo-check-all-features", test_arg = ["--help"] }
-install_crate_args = ["--version", "^1.9"]
```
Comment (sffc) [inline tools/make/tests.toml:11]: Question: why did you remove the `install_crate_args`; IIRC it helps prevent cases where people have old versions of the tool installed

**Review by sffc — APPROVED** [review]

File: tools/make/tests.toml (hunk 1 above)
Comment (Manishearth) [reply tools/make/tests.toml:11]: oh, yeah, Rob's changes changed them and I removed them because they were referencing a hash. I'll reintroduce them

Comment (Manishearth) [reply tools/make/tests.toml:11]: oh, yeah, Rob's changes changed them and I removed them because they were referencing a hash. I'll reintroduce them

**Comment by Manishearth** [comment]
force merging since branch protection rules need to be updated
